src/main/java/com/google/devtools/build/lib/sandbox/SandboxOptions.java

```diff
@@ -369,6 +373,52 @@ public ImmutableSet<Path> getInaccessiblePaths(FileSystem fs) {
               + " Requires cgroups v1 or v2 and permissions for the users to the cgroups dir.")
   public int memoryLimitMb;
 
+  @Option(
+      name = "experimental_sandbox_limits",
+      defaultValue = "null",
+      documentationCategory = OptionDocumentationCategory.EXECUTION_STRATEGY,
+      effectTags = {OptionEffectTag.EXECUTION},
+      converter = ResourceConverter.AssignmentConverter.class,
+      allowMultiple = true,
+      help =
+          "If > 0, each Linux sandbox will be limited to the given amount"
+              + " for the specified resource. Overrides --experimental_sandbox_memory_limit_mb."
+              + " Requires cgroups v1 or v2 and permissions for the users to the cgroups dir.")
+  public List<Map.Entry<String, Double>> limits;
+
+  public ImmutableMap<String, Double> getLimits() {
+    return ImmutableMap.<String, Double> builder()
+        .put("memory", (double) memoryLimitMb)
+        .putAll(limits)
+        .buildKeepingLast();
+  }
+
+  @Option(
+    name = "incompatible_use_old_cgroup_implementation",
```

Comment:
nit: name as `incompatible_use_new_cgroup_implementation` and with the default value as false.

(use of the new cgroup implementation makes it incompatible because it takes the limits from `--experimental_sandbox_limits` instead of `--experimental_sandbox_memory_limit_mb`; the old implementation isn't incompatible).